### PR TITLE
Support for multiple playlists

### PR DESCRIPTION
**Pull request by AndreyPavlenko**

### CODE REVIEW

**Comment by sei4eto** [comment]
Hello!
How can I get this merged version with multi playlist support?
Thanks!

**Comment by AndreyPavlenko** [comment]
You can find it here - https://github.com/AndreyPavlenko/pvr.iptvsimple

**Comment by sei4eto** [comment]
Thanks.
I follow provided instructions, everything was done, but where i can find the .zip made from this repo?
Thanks

**Comment by AndreyPavlenko** [comment]
Do you need a zip with sources or binaries?

**Comment by sei4eto** [comment]
i need a iptv simple.zip to install it in kodi (instead of original iptv.pvr in kodi). I need multiple m3u playlist support.
Thanks

**Comment by AndreyPavlenko** [comment]
Then you need to compile the sources. I have only packages for Ubuntu - https://launchpad.net/~aap/+archive/ubuntu/kodi-addons/ .
What's your operating system?

**Comment by sei4eto** [comment]
Hi!
I have multiple OS - Android TV, Raspberry Pi2 , Ios9.
Any help will be very appreciate.
Thanks

**Comment by AndreyPavlenko** [comment]
Unfortunately I don't have any of the above systems and I can't build the addon for you.
Please try to build it on your own following the build instruction - https://github.com/kodi-pvr/pvr.iptvsimple/blob/master/README.md .

**Comment by sei4eto** [comment]
I've already made it. The build process is finished. But now I see the builded files and folders,but I can't figure out how to install builded files into Kodi.
Sorry for noob questions.
Thanks!

**Comment by AndreyPavlenko** [comment]
make package-addons

**Comment by AndreyPavlenko** [comment]
To build for the Jarvis release of Kodi:

1. git clone https://github.com/xbmc/xbmc.git -b Jarvis
2. git clone https://github.com/AndreyPavlenko/pvr.iptvsimple.git -b Jarvis
3. cd pvr.iptvsimple && mkdir build && cd build
4. cmake -DADDONS_TO_BUILD=pvr.iptvsimple -DADDON_SRC_PREFIX=../.. -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=../../xbmc/addons -DPACKAGE_ZIP=1 ../../xbmc/project/cmake/addons
5. make
6. make package-addons

**Comment by sei4eto** [comment]
Thank you!
This is exactly what I’ve looked for!!!
I succeed to make .zip for OSX. Here is compiled .zip https://www.dropbox.com/s/v38kr8rjertdzmq/addon-pvr.iptvsimple-2.0.0.zip?dl=0 <https://www.dropbox.com/s/v38kr8rjertdzmq/addon-pvr.iptvsimple-2.0.0.zip?dl=0>
Unforunately i can’t make it for Android nor for iOS. I get “cannot instal from zip, located ….” error. 
Any further suggestions?  Shall i make something to make .zip compatible with Android and iOS ?
p.s. When this fork will be in the main pvr repo?
Thanks again.

**Comment by AndreyPavlenko** [comment]
You need to build the addon for each platform. To be honest I never build the addon neither for android nor for iOS and I can't help you with this. Try to google.

I don't know when (if ever) it will be merged to the main repo.

**Comment by catbaxter** [comment]
Hi Andrey, first - great job, hope to see the merged code very soon! Stupid question - do you know how to modify plugin sources in order to support the switch to EPG programm in archive? For example - a playlist supports 7 day archive and you can start any programm just add utc_start and utc_stop parameters, for example: 

playlist.m3u8?utc_start=1454587200&utc_stop=1454587800

It shoud happens in EPG dialog (using 'switch' button), but someone should modify url for EPG programm. Do you know the trick? :)

**Comment by AndreyPavlenko** [comment]
Hi. I'm not sure I understood the requirement. Which part is expected to handle the utc_start and utc_stop parameters?

File: .gitignore [hunk 1]
```diff
@@ -12,7 +12,16 @@ debian/kodi-pvr-iptvsimple.substvars
 debian/kodi-pvr-iptvsimple/
 debian/tmp/
 obj-x86_64-linux-gnu/
+CMakeCache.txt
```
Comment (Jalle19) [inline .gitignore]: The idea is that you build "out of tree" in a separate `build` directory, then you only need to ignore that directory. Can you do that?

File: pvr.iptvsimple/resources/language/resource.language.en_au/strings.po [hunk 2]
```diff
@@ -87,3 +87,31 @@ msgstr "Channels Logos Base URL"
 msgctxt "#30042"
 msgid "Ignore"
 msgstr "Ignore"
+
```
Comment (Jalle19) [inline pvr.iptvsimple/resources/language/resource.language.en_au/strings.po]: You should only update `resource.language.en_gb/strings.po`, the rest are handled by the translation system. So drop all other language changes except for that file.

Comment (AndreyPavlenko) [reply pvr.iptvsimple/resources/language/resource.language.en_au/strings.po]: Reverted au, nz and us files.

File: .gitignore (hunk 1 above)
Comment (AndreyPavlenko) [inline .gitignore]: I believe most people, including me, build from within the tree. This is also true for IDEs.
Are there any issues with this commit?

**Comment by afedchin** [comment]
I have a veto for this. I don't like an idea support multiple playlists, I don't like an idea to support this monstrous settings.xml

**Comment by AndreyPavlenko** [comment]
Hm... What do you have against multiple playlists? This is a quite often requested feature.
The monstrous settings.xml is probably not the best solution, but an alternative solution would be to implement the settings programmatically. Besides, the settings.xml file is generated automatically by the shell script.

**Comment by Jalle19** [comment]
I don't like the huge settings.xml either, it all feels a bit hacky. Plus there are zero legitimate use cases for this as far as I can tell.

**Comment by AndreyPavlenko** [comment]
This is indeed hacky, I agree, but I didn't find a better solution to implement a dynamic menu in pure xml.
There are so many iptv providers in the internet. It's absolutely legitimate to use multiple providers, for example, to watch tv from different countries.

**Comment by Jalle19** [comment]
I meant legitimate as in "not most likely illegal".

**Comment by Jalle19** [comment]
That was supposed to read "not legitimate as in most likely illegal"

**Comment by ChadiEM** [comment]
I'm not against the idea of multiple playlists. However, the duplicated settings in settings.xml is complicated and tends to be error-prone (ie if you make a change, but forget to change all of them). How about something more simple? I had in mind a comma separated list of playlists, all sharing the same settings. Similarly for EPG. What do you think?

**Comment by AndreyPavlenko** [comment]
I agree that this is error-prone, so I've created the shell script which generates the settings - https://github.com/AndreyPavlenko/pvr.iptvsimple/blob/63e3f644ebc85feabc1170ec5b3d3bbe33aed69e/pvr.iptvsimple/resources/gensettings.sh. All changes must be done here but not in the xml file.

Comma separated list easier for dev, but harder for user. Besides, playlist is not just only a single url but also epg, logo, epg shift etc. In addition, in my implementation it's also possible to assign prefixes for channel/group names from different playlists. For example, I add the provider name prefix to the group names.

**Comment by blablack** [comment]
I actually support this pull request and I can "show" a legal reason to use it - I am french and I'm living abroad:
- I use camembert.tv for french channels (100% legal and paid for)
- I use iBox.ie for irish channels (100% legal and paid for)

**Comment by VVi3ard** [comment]
Hi. Im use 2 m3u auto refresh (Http://) playlist from two HTTP services (TV and RADIO).  Also im have small playlist local (other radio).

I try input two urls split ";". Eample: http://cooliptv/playlist.m3u; http://coolradio/playlist.m3u but it not work.

**Comment by AndreyPavlenko** [comment]
Semicolon separated URLs are not supported. Use this https://github.com/AndreyPavlenko/pvr.iptvsimple addon if you need support for multiple URLs.

**Comment by nsenica** [comment]
Hey

Any chance we will see support multiple playlists in the official release ?

A big +1 from me.

**Comment by Jalle19** [comment]
Probably not

**Comment by Woody0x2A** [comment]
First of all, thanks Andrey!
Second, I strongly support this feature.
So really, why not?
What needs to be done to make this official?

**Comment by CanisLupus03** [comment]
It would be nice, if i can have multiple playlists.

I am using an AVM Fritz DVB-C Repeater, which makes 3 Playlists, one for SD TV, one for HD TV and one for Radio, so it would be great if i can use alle of them.

**Comment by phunkyfish** [comment]
I have a friend who is from a multilingual household. His current set up is 3 kodi boxes per TV (6 in total) to support the 3 legitimate IPTV providers they use. I think a limited set of providers (maybe 3 or 4) should be acceptable to support legitimate use cases. Having that done in cleaner fashion would be nice.

**Comment by afedchin** [comment]
> I have a friend who is from a multilingual household. His current set up is 3 kodi boxes per TV (6 in total) to support the 3 legitimate IPTV providers they use. I think a limited set of providers (maybe 3 or 4) should be acceptable to support legitimate use cases. Having that done in cleaner fashion would be nice.

your friend can select another solutions, tvheadend as an example.

**Comment by phunkyfish** [comment]
> your friend can select another solutions, tvheadend as an example.

True, but for the less technically inclined they tend to stick with this addon.

**Comment by matthuisman** [comment]
@phunkyfish
Try this:
https://www.matthuisman.nz/2019/02/iptv-merge-kodi-add-on.html

**Comment by phunkyfish** [comment]
Nice, I will pass it on.

**Comment by bam80** [comment]
What is the current status of this?
It would be fine to have ability for multiple EPG sources at once, without merging them in one file beforehand.

**Comment by phunkyfish** [comment]
Closing as there are no further updates on Leia.

**Pull request closed without merging**